nucypher/blockchain/eth/agents.py

```diff
@@ -185,17 +193,20 @@ def mint(self, node_address) -> Tuple[str, str]:
         self.blockchain.wait_for_receipt(mint_txhash)
         return mint_txhash
 
-    def collect_staking_reward(self, collector_address) -> str:
-        """Withdraw tokens rewarded for staking."""
-
-        token_amount = self.contract.functions.minerInfo(collector_address).call()[0]
-        staked_amount = max(self.contract.functions.getLockedTokens(collector_address).call(),
-                            self.contract.functions.getLockedTokens(collector_address, 1).call())
-
-        collection_txhash = self.contract.functions.withdraw(token_amount - staked_amount).transact({'from': collector_address})
+    @validate_checksum_address
+    def calculate_staking_reward(self, checksum_address: str) -> int:
+        token_amount = self.contract.functions.minerInfo(checksum_address).call()[0]
```

Comment:
This function is a little confusing to me.  So, the "staking reward" is the total token amount *less* the staked amount?

Reply:
Hm - Yes, you're right.  There may be a better way to calculate what rewards are available for withdraw.   This is some older code at this point.  I need to give it some more thought.  Perhaps a future PR? @szotov

Reply:
Staker can't just put some tokens in the contract without locking it, so tokens is stake plus reward plus unlocked tokens. Withdraw only reward is almost impossible. Calculation in function is right, maybe better to have more general name

Reply:
>  any amount in my wallet in excess of the staked amount is a reward

This is not your wallet, really, it's a quantity of locked-tokens "held" in-escrow, If I understand correctly.

Reply:
`self.contract.functions.minerInfo` - roger.

Reply:
Yes - This is more or less `MinerEscrow.minerInfo[address]`